src/providers/index.js

```diff
@@ -0,0 +1,51 @@
+import axios from "axios"
+import jwtDecode from "jwt-decode"
+
+import buildCharacterProvider from "./character"
+import buildAuthenticationProvider from "./authentication"
+import buildStorageProvider from "./storage"
+import buildJwtProvider from "./jwt"
+
+const baseURL = "http://localhost:3050" // TODO: FIX TO USE ENV process.env.REACT_APP_MAHABHARATH_API
+
+const http = axios.create({})
+const httpAuthenticated = axios.create({})
+
+const storageProvider = buildStorageProvider()
+const authenticationProvider = buildAuthenticationProvider({ http, baseURL, storageProvider })
```

Comment:
Here, I'm using dependency injection, the name is scary, but is a simple concept.

When using the javascript [`forEach`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach#syntax) function you can pass a `callbackFn`, this is a function that will the executed on every iteration of an array, so in JS functions can be passed as arguments to functions, and this is what we're doing here, 
```js
const http = axios.create({})
const httpAuthenticated = axios.create({})
```
here, we are creating two instances of axios, one will be used for routes who needs authentication, and the other for routes who doesn't need authentication
Now, the character information doesn't need the user to login, but if in the future you wanted to change this, you could do:
```js
const characterProvider = buildCharacterProvider({ http, baseURL }) // change this
const characterProvider = buildCharacterProvider({ http: httpAuthenticated, baseURL }) // for this
```

Reply:
yeah got the idea.